### ISSUE TITLE
BUG: Unable to retrieve MagicMirror modules. Is your internet connection up?

### ISSUE THREAD

**Issue by td00** [Bee-Mar/mmpm#53]
**Describe the Bug**

Is this an issue with the CLI or GUI?
mmpm doesn't start

Is this something that should be put in the Wiki?
if it's not a weird bug on my system, maybe enhancing the setup guide would be a nice idea.

When installing mmpm via python3 on a Ubuntu 20.04.01 LTS with a MagicMirror via Docker-Compose, the mmpm won't start after installing and configuring.

**To Reproduce**
1. Install as told in the guidelines
2. Configuring with `mmpm --guided-setup`
3. Try to do something (p.e. `mmpm log`)
4. See the error

**Expected behavior**
Able to use the software

**Operating System, Python3 Version, and Web Browser**\
(Please complete the following information)

- output from `cat /etc/os-release`
	- NAME="Ubuntu"
VERSION="20.04.1 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.1 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal

- output from `python3 --version`
	- Python 3.8.5

**MMPM Environment Settings**

- output from `mmpm env`
	- + Initializing MagicMirror 3rd party packages database ✘
FATAL: Unable to retrieve MagicMirror modules. Is your internet connection up?

**MMPM Log Files**

- Attach the file produced `Download MMPM log files` within the Control Center of the GUI or from the  `mmpm log --zip`
	- unable to produce. Same error.
	- Content of mmpm-cli-interface.log is repeatedly:
	 
```
2020-10-26 20:15:12,804.804 CRITICAL utils - fatal_msg: Unable to retrieve MagicMirror modules. Is your internet connection up?
Unable to retrieve MagicMirror modules. Is your internet connection up?
```
**MagicMirror Config**

- your MagicMirror `config.js`
	- not existing

**Comment by td00** [comment]
ftr: yes! My internet connection is up.
No proxy, no cgn, no DNS errors, everything else works.

**Comment by Bee-Mar** [comment]
You say your MagicMirror config.js doesn’t exist, is that correct? If that’s the case, this feature will not work since you can’t add the configuration for the MMPM MagicMirror module

**Comment by Bee-Mar** [comment]
This is most likely an issue with the MagicMirror 3rd party modules page. Someone probably altered the markdown/html which caused parsing to fail

**Comment by td00** [comment]
Sorry.
Yeah I definitly have a config.js for my magicmirror.
Don't know where my head was when writing this issue... X.x

**Comment by Bee-Mar** [comment]
@td00 so, this isn't related to the magicmirror modules wiki. Have you tried installing MMPM in a virtualenv?

**Comment by starlord1980** [comment]
Same here then I guess, just trying this out and run out of ideas:

> pi@raspberrypi:~ $ mmpm env
{
  "MMPM_MAGICMIRROR_ROOT": "/home/pi/MagicMirror",
  "MMPM_MAGICMIRROR_URI": "http://127.0.1.1:8080",
  "MMPM_MAGICMIRROR_PM2_PROCESS_NAME": "",
  "MMPM_MAGICMIRROR_DOCKER_COMPOSE_FILE": ".",
  "MMPM_IS_DOCKER_IMAGE": false
}

I believe I've set this correctly according to the instructions which state:
- MMPM_MAGICMIRROR_URI is correct
- ipWhitelist in your MagicMirror config.js includes 0.0.0.0, or the ipWhitelist is an empty array.
Presumably this is all correct?

> var config = {
	address: "0.0.0.0", 	
	port: 8080,
	basePath: "/", 	
	ipWhitelist: []

**Comment by td00** [comment]
> @td00 so, this isn't related to the magicmirror modules wiki. Have you tried installing MMPM in a virtualenv?

Sorry, your reply got burried.
I've installed without an venv.
Just via the pip install command and then tried to use it.

On the difference to @starlord1980 i get the same message when using "mmpm env" so i'll assume, that i've don't have the same bug as him.

**Comment by td00** [comment]
It seems, like there's was a problem using a previous installed MagicMirror version.
I've removed the MagicMirror Version installed & installed a new Version via mmpm.
After the installation via mmpm & restoring the config & modules, everything works fine.

It seems like the error, produced in this constellation was just misleading.
Not quite sure how to reproduce this error.
Tried it on a similar machine, with a backup from the previous not working "prod" system and there it worked like a charm.

Seems to be a bug on my system.

Thanks for looking into it.

cheers :)
